sklearn/utils/_mocking.py

```diff
@@ -148,8 +148,10 @@ def _check_X_y(self, X, y=None, should_be_fitted=True):
         ----------
         X : array-like of shape (n_samples, n_features)
             The data set.
+            `X` is checked only if `check_X` is not `None` (default is None).
```

Comment:
I think it's better to have this closer to the actual parameter that can be `None`:

https://github.com/scikit-learn/scikit-learn/blob/c2f5782b20751a258f4bbbcc6c201c809369669e/sklearn/utils/_mocking.py#L71-L73

I also prefer this wording:

> The callable used to validate `X` and `y`. These callable should return
  a bool where `False` will trigger an `AssertionError`. If `None`, then
  then the data is not validated.

Reply:
Take a look at https://github.com/scikit-learn/scikit-learn/pull/25997/commits/d632b8371e0c76aae56436e28bdee9c711c4237d. I've added the suggestion to l. 72 and further `None` documentation to the related `check_y_params` in l. 77.